kasa/smartdevice.py

```diff
@@ -325,10 +329,16 @@ async def _modular_update(self, req: dict) -> None:
             )
             self.add_module("emeter", Emeter(self, self.emeter_type))
 
-        for module in self.modules.values():
+
+        for module_key in self.modules:
+            module = self.modules[module_key]
             if not module.is_supported:
                 _LOGGER.debug("Module %s not supported, skipping" % module)
                 continue
+            elif self.model in MODULE_EXCLUSIONS and ( module_key in MODULE_EXCLUSIONS[self.model] ):
+                _LOGGER.debug("Module %s is excluded for %s, skipping" % (module, self.model))
+                continue
```

Comment:
```suggestion

            modules_to_skip = MODULE_EXCLUCIONS.get(self.model, [])
            if self.model in modules_to_skip:           
                _LOGGER.debug("Module %s is excluded for %s, skipping" % (module, self.model))
                continue
```
No need for `elif` here. My suggestion would be to keep it simple using something like this.

Reply:
Great suggestion - thanks!